# Abschlussprojekt - Dataset Kiva

# Teil - Visualisierung

In [1]:
# benötigte bib importieren

import pandas as pd 
import plotly.express as px 
import numpy as np 
from plotly.offline import init_notebook_mode, iplot
init_notebook_mode(connected=True)
import seaborn as sns 

In [2]:
# datensatz laden mit pickle

df = pd.read_pickle('df.pkl')      
df.head(5)

funded_amount  loan_amount             activity          sector  \
0          300.0        300.0  Fruits & Vegetables            Food   
1          575.0        575.0             Rickshaw  Transportation   
2          150.0        150.0       Transportation  Transportation   
3          200.0        200.0           Embroidery            Arts   
4          400.0        400.0           Milk Sales            Food   

                                                 use country_code   country  \
0            To buy seasonal, fresh fruits to sell.            PK  Pakistan   
1  to repair and maintain the auto rickshaw used ...           PK  Pakistan   
2  To repair their old cycle-van and buy another ...           IN     India   
3  to purchase an embroidery machine and a variet...           PK  Pakistan   
4                           to purchase one buffalo.           PK  Pakistan   

  currency  term_in_months  lender_count borrower_genders repayment_interval  \
0      PKR              12            12           female          irregular   
1      PKR              11            14            group          irregular   
2      INR              43             6           female             bullet   
3      PKR              11             8           female          irregular   
4      PKR              14            16           female            monthly   

   gender  borrower_gender_count  
0  female                      1  
1  female                      2  
2  female                      1  
3  female                      1  
4  female                      1

**Welche Effizienz hat unsere Plattform?**

In [3]:
#Daten aggregieren nach funded_amount und loan_amount
grouped_df = df.groupby('sector')[['funded_amount', 'loan_amount']].sum().reset_index().sort_values('funded_amount', ascending = False)
grouped_df

sector  funded_amount  loan_amount
0      Agriculture    133770635.0  143067875.0
6             Food    115089210.0  121606150.0
11          Retail     90864855.0   98122900.0
12        Services     42966085.0   48057450.0
2         Clothing     34805270.0   37300925.0
4        Education     30044920.0   30968525.0
8          Housing     21377100.0   23663950.0
10    Personal Use     14189775.0   14955350.0
1             Arts     11821055.0   12240325.0
13  Transportation      9842500.0   11066900.0
7           Health      9172250.0    9843400.0
3     Construction      6321515.0    6695150.0
9    Manufacturing      5356650.0    5447900.0
5    Entertainment      1023095.0    1389150.0
14       Wholesale       918900.0     995200.0

In [4]:
# Berechnen  den Prozentsatz von funded_amount im Verhältnis zu loan_amount und erstellen neue Spalte 
grouped_df['percentage'] = round((grouped_df['funded_amount'] / grouped_df['loan_amount']) * 100)
grouped_df

sector  funded_amount  loan_amount  percentage
0      Agriculture    133770635.0  143067875.0        94.0
6             Food    115089210.0  121606150.0        95.0
11          Retail     90864855.0   98122900.0        93.0
12        Services     42966085.0   48057450.0        89.0
2         Clothing     34805270.0   37300925.0        93.0
4        Education     30044920.0   30968525.0        97.0
8          Housing     21377100.0   23663950.0        90.0
10    Personal Use     14189775.0   14955350.0        95.0
1             Arts     11821055.0   12240325.0        97.0
13  Transportation      9842500.0   11066900.0        89.0
7           Health      9172250.0    9843400.0        93.0
3     Construction      6321515.0    6695150.0        94.0
9    Manufacturing      5356650.0    5447900.0        98.0
5    Entertainment      1023095.0    1389150.0        74.0
14       Wholesale       918900.0     995200.0        92.0

In [5]:
# Datentyp für der Spalte percentage ändern
grouped_df['percentage']= grouped_df['percentage'].astype(int)

In [6]:
# Barplot mit Plotly Express erstellen 


fig = px.bar(
    grouped_df,
    x='sector',
    y=['funded_amount', 'loan_amount'],
    text='percentage',
    barmode='group', #zwei Bars in der Nähe 
    title='Summe der Projekte  nach Sektor',
    template = 'ggplot2',
    labels={'sector': 'Sector', 'value': 'Summe'}
)
fig.update_traces(textfont_size=13, textangle=0, textposition="outside")  
fig.data[1].text = [] #keinen Text für bar loan_amount
# Plot anzeigen lassen
fig.show()

- Das Bardiagramm zeigt einen hohen Prozensatz der Plattfromeffizienz. In fast jedem Sektor außer Services, Transportation und Entertaiment,  beobachten wir, dass der ausgezahlte Betrag mehr als 90 Prozent des erforderlichen Betrags ist. 
- In Sektoren **Education** und **Arts** - 97%, **Manufacturing** - 98%

In [7]:
grouped_df['percentage'].mean()

92.2

- Der durchschnittliche Finanzierungsanteil liegt bei 92.2 Prozent.
- Dieser Prozentsatz beweist, dass die Platform funktioniert und Unternehmen und Benutzern einen Gewinn bringt.

**Fragen**
- Ist es möglich, diesen Prozentsatz zu erhöhen, indem neue Gläubiger angezogen werden?
- Wie viel kostet ein neuer Kreditgeber für Geschäft? 
- Sollte Geschäftsleitung auf die **Education** , **Arts** und **Manufacturing** achten, da es in diesen Bereichen nicht viele Projekte gibt, aber die Menschen bereit sind, in sie zu investiren? 

**Zusammenhang zwischen verschidenen Metriken**

In [8]:
# Dataframe für Scatterplot ohne Projekte , für den kein Geld gesammelt wird und ohne das Projekt für 100000 USD 

df_for_scatter = df[(df['funded_amount'] < 100000) & (df['funded_amount'] != 0)]

In [14]:
#Scatterplot mehr als 25MB  wurde als png gespeichert.
# Scatterplot mit Plotly Express erstellen 
#fig = px.scatter(
#    df_for_scatter,
#    x='lender_count',
#    y='funded_amount',
#    color = 'repayment_interval',
#    size = 'term_in_months',
#    template = 'ggplot2',
#    title='Zusammenhang zwischen Kreditgeberanzahl  und Betrag',
#   opacity=0.5,  
#   labels={'lender_count': 'Kreditgeberanzahl', 'funded_amount': 'Betrag'},
                 )
#fig.update_layout(
#    title=dict(
#        x=0.5,  #  den Titel horizontal zentrieren
#        y=0.9   # die vertikale Position des Titels anpassen
#    )
#)
# Plot anzeigen lassen
#fig.show()

IndentationError: unexpected indent (3197829225.py, line 13)

- Direkte Abhängigkeit der Anzahl der Kreditgeber von der Kreditsumme. Je höher die  Kreditsumme ist , desto mehr Kreditgeber. 
- Es gibt keine Zusammenhang zwischen Betrag und der Dauer in Monaten. Eine große Anzahl von Monaten findet sich auch bei kleinen und großen Kreditsummen.
- Es gibt keine großen Investoren. Gibt es Gründe? 
- **Darin besteht die Möglichkeit für Wachstum.** Je schneller die Kreditnehmer das Darlehen erhalten, desto schneller wird das Unternehmen einen Gewinn erzielen. 

- Es gibt keine Zusammenhang zwischen Betrag und der Dauer in Monaten. Eine große Anzahl von Monaten findet sich auch bei kleinen und großen Kreditsummen.

**Die Länder**

In [10]:
# aggregierte Daten
df_2 = df.groupby(by='country', as_index=False).agg(count=("country","count"))
df_2_sort =  df_2.sort_values(by = 'count', ascending = False).head(10) 
df_2_sort

country   count
59  Philippines  160441
34        Kenya   75825
22  El Salvador   39875
11     Cambodia   34836
54     Pakistan   26857
58         Peru   22233
15     Colombia   21995
78       Uganda   20601
71   Tajikistan   19580
20      Ecuador   13521

In [11]:
# Barplot mit Plotly Express erstellen 

barplot2 = px.bar(data_frame=df_2_sort, 
                 x='country',
                 y='count', 
                 labels={'country':'Land', 'count':'Anzahl der Projekte'},
                 title='Anzahl der Projekte je Land',
                 template='ggplot2')
# Plot anzeigen lassen
barplot2.show()

- In den Philippinen sind die meisten Projekte - mehr als 160.000

In [12]:
# Daten aggregieren nach Summe und Land

df_3 = df.groupby('country')['funded_amount'].sum().reset_index().sort_values('funded_amount', ascending = False)

# Neue Spalte mit  ISO Alpha-3 Codes für plotly express
df_3['code'] = ["PHL", "KEN", "PER", "PRY", "SLV", "USA", "KHM", "BOL", "RWA", "ECU", "UGA", "TJK", "VNM", "COL", "PAK", "PSE", "LBN", "ARM", "COD", "GTM", "NIC", "MEX", "MLI", "SEN", "KGZ", "TZA", "IND", "HND", "WSM", "GHA", "IDN", "JOR", "SLE", "ZWE", "GEO", "MMR", "BFA", "AZE", "HTI", "IRQ", "BDI", "ALB", "TLS", "DOM", "CRI", "MOZ", "NGA", "YEM", "XKX", "MWI", "MNG", "UKR", "TGO", "MDG", "LBR", "LAO", "ZMB", "EGY", "CMR", "COD", "TUR", "ISR", "MDA", "BRA", "ZAF", "SUR", "BEN", "SLB", "THA", "CHN", "LSO", "NPL", "PRI", "PAN", "SOM", "SSD", "BLZ", "CHL", "CIV", "VCT", "NAM", "BTN", "MRT", "AFG", "VUT", "GUM", "VIR"]
df_3

country  funded_amount code
59     Philippines     54476375.0  PHL
34           Kenya     32248405.0  KEN
58            Peru     30394850.0  PER
57        Paraguay     29412700.0  PRY
22     El Salvador     23357725.0  SLV
..             ...            ...  ...
44      Mauritania        15000.0  MRT
0      Afghanistan        14000.0  AFG
81         Vanuatu         9250.0  VUT
25            Guam          395.0  GUM
83  Virgin Islands            0.0  VIR

[87 rows x 3 columns]

In [13]:
#Karte erstellen
fig = px.choropleth(data_frame=df_3,
                    locations=df_3['code'], 
                    color=df_3['funded_amount'],
                    color_continuous_scale=px.colors.diverging.Geyser, #Farbschema auswählen
                    #color_continuous_scale='Viridis', 
                    #color_continuous_scale=px.colors.qualitative.Bold,
                    hover_data=['country'],
                    title='Kreditsumme je Land')
                    #color_continuous_label='Summe')
fig.update_layout(
    title=dict(
        x=0.5,  #  den Titel horizontal zentrieren
        y=0.9)  # die vertikale Position des Titels anpassen
               )
#Karte darstellen
fig.show() 

- Kiva ist in 87 Ländern verbreitet.
- Die größte Summe der ausgegebenen Kredite auf den Philipinen. 
- In Afrika ist die gröste Summe in Kenia. 
- Kiva ist auch in Lateinamerika beliebt - Peru, Bolivia, Paraguay.
- Die meisten Kreditnehmer kommen aus Entwicklungsländern